seqr/views/apis/anvil_workspace_api.py

```diff
@@ -193,29 +194,79 @@ def create_project_from_workspace(request, namespace, name):
 
     project = create_model_from_json(Project, project_args, user=request.user)
 
+    _trigger_add_workspace_data(project, pedigree_records, request.user, request_json)
+
+    return create_json_response({'projectGuid': project.guid})
+
+
+@anvil_auth_and_policies_required
+def add_workspace_data(request, project_guid):
+    """
+    Add data from an AnVIL workspace.
+
+    :param request: Django request object
+    :param project_guid: Django request object
+    :return success if no exceptions
+
+    """
+    project = get_project_and_check_permissions(project_guid, request.user, can_edit=True)
+    check_workspace_perm(request.user, CAN_EDIT, project.workspace_namespace, project.workspace_name, can_share=True,
+                         meta_fields=['workspace.bucketName'])
+
+    request_json = json.loads(request.body)
+
+    missing_fields = [field for field in ['uploadedFileId', 'fullDataPath', 'vcfSamples'] if not request_json.get(field)]
+    if missing_fields:
+        error = 'Field(s) "{}" are required'.format(', '.join(missing_fields))
+        return create_json_response({'error': error}, status=400, reason=error)
+    request_json['genomeVersion'] = project.genome_version
+    request_json['sampleType'] = Sample.objects.filter(individual__family__project=project)[0].sample_type
+
+    # Parse families/individuals in the uploaded pedigree file
+    json_records = load_uploaded_file(request_json['uploadedFileId'])
+    pedigree_records, _ = parse_pedigree_table(json_records, 'uploaded pedigree file', user=request.user, fail_on_warnings=True)
+
+    indivIds = {ped['individualId'] for ped in pedigree_records}
+    pedigree_records += [ped for ped in get_project_active_pedigrees(project) if ped['individualId'] not in indivIds]
```

Comment:
When getting the previous samples, we filter the samples with `dataset_type=Sample.DATASET_TYPE_VARIANT_CALLS` as your suggestion. Does it mean we don't support other dataset type like `SV` or `MITO`?

Reply:
no, we do not support other data types for AnVIL users in general, and the automated loading pipeline kicked off by this request is variant loading. It will be a lot of engineering work on our end to make SV and MITO loading available for AnVIL data, which is out of scope of this feature as we need to add the loading to airflow and add better file format documentation and validation before we can make it broadly available